crates/red_knot_python_semantic/src/types.rs

```diff
@@ -108,77 +108,6 @@ fn widen_type_for_undeclared_public_symbol<'db>(
 
 /// Infer the public type of a symbol (its type as seen from outside its scope).
 fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
-    #[salsa::tracked]
-    fn symbol_by_id<'db>(
-        db: &'db dyn Db,
-        scope: ScopeId<'db>,
-        is_dunder_slots: bool,
-        symbol_id: ScopedSymbolId,
-    ) -> Symbol<'db> {
-        let use_def = use_def_map(db, scope);
-
-        // If the symbol is declared, the public type is based on declarations; otherwise, it's based
-        // on inference from bindings.
-
-        let declarations = use_def.public_declarations(symbol_id);
-        let declared = symbol_from_declarations(db, declarations);
```

Comment:
We'll need a salsa query somewhere because evaluating the visibility constraint access `.node_ref` but we can consider moving the salsa query closer to where it is needed?